src/expenses/expensesService.js

```
@@ -0,0 +1,96 @@
+import { v4 as uuidv4 } from 'uuid';
+
+const expenses = [];
+
+export function getAllExpenses(userId, categories, dateFrom, dateTo) {
+  let filteredExpenses = expenses;
+
+  if (typeof userId === 'number') {
+    filteredExpenses = filteredExpenses.filter((e) => e.userId === userId);
```

`getAllExpenses` checks `typeof userId === 'number'`. If `userId` comes from query params it will likely be a string. Either normalize/convert the incoming param (recommended) or accept both string and number forms here. Otherwise filtering by userId will silently fail.